### PR TITLE
refactor: adapt blocks and clean some code

### PR DESCRIPTION
**Pull request by Draglis**

### CODE REVIEW

File: src/main/java/cn/nukkit/block/BlockBamboo.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public int onUpdate(int type) {
             level.useBreakOn(this, null, null, true);
         } else if (type == Level.BLOCK_UPDATE_RANDOM) {
             Block up = up();
-            if (getAge() == 0 && up.getId() == AIR && level.getFullLight(up) >= BlockCrops.MINIMUM_LIGHT_LEVEL && ThreadLocalRandom.current().nextInt(3) == 0) {
```
Comment (CoolLoong) [inline src/main/java/cn/nukkit/block/BlockBamboo.java:58]: what is `upisAir()`

File: src/main/java/cn/nukkit/registry/BlockRegistry.java [hunk 2]
```diff
@@ -85,12 +85,12 @@ public void init() {
         register(BIRCH_DOOR, BlockBirchDoor.class);// done.
         register(BIRCH_FENCE, BlockBirchFence.class);// done.
         register(BIRCH_FENCE_GATE, BlockBirchFenceGate.class);// done.
-        register(BIRCH_HANGING_SIGN, BlockBirchHangingSign.class);// done.
+        register(BIRCH_HANGING_SIGN, BlockBirchHangingSign.class);
```
Comment (CoolLoong) [inline src/main/java/cn/nukkit/registry/BlockRegistry.java:88]: Why are these annotation be deleted?

**Review by CoolLoong — CHANGES_REQUESTED** [review]

File: src/main/java/cn/nukkit/block/BlockFlowerPot.java [hunk 3]
```diff
@@ -126,7 +126,7 @@ public Item getFlower() {
 
 
     public boolean setFlower(@Nullable Item item) {
-        if (item == null || item.getId() == AIR) {
+        if (item == null || item.isAir()) {
```
Comment (CoolLoong) [inline src/main/java/cn/nukkit/block/BlockFlowerPot.java:129]: `Item` class cant find `isAir`

**Review by CoolLoong — CHANGES_REQUESTED** [review]

File: src/main/java/cn/nukkit/block/BlockFlowerPot.java (hunk 3 above)
Comment (CoolLoong) [reply src/main/java/cn/nukkit/block/BlockFlowerPot.java:129]: and you can use `isNull`

**Review by CoolLoong — APPROVED** [review]

File: src/main/java/cn/nukkit/block/BlockBamboo.java (hunk 1 above)
Comment (Draglis) [reply src/main/java/cn/nukkit/block/BlockBamboo.java:58]: Forget to add '.' between up and isAir()

**Pull request merged**